### PR TITLE
remove duplicated examples, and rename demo to v1_api_demo

### PR DESCRIPTION
**Pull request by luotao1**
1. solve #2350  with comments of #2353 
2. add [readme.md](https://github.com/luotao1/Paddle/blob/v1_demo/v1_api_demo/README.md) for v1_api_demo
3. remove unused *.pyc for VAE demo.

### CODE REVIEW

File: demo/mnist/vgg_16_mnist.py [hunk 1]
```diff
@@ -1,50 +0,0 @@
-# Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved
```
Comment (lcy-seso) [inline demo/mnist/vgg_16_mnist.py]: This example is not in PaddleBook either.

File: demo/mnist/light_mnist.py [hunk 2]
```diff
@@ -1,79 +0,0 @@
-# Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved
```
Comment (luotao1) [inline demo/mnist/light_mnist.py]: Done

File: demo/mnist/vgg_16_mnist.py (hunk 1 above)
Comment (luotao1) [inline demo/mnist/vgg_16_mnist.py]: Done

**Comment by luotao1** [comment]
remain mnist demo currently.

**Pull request merged**